plugins/aws/provisioner.go

```diff
@@ -2,42 +2,155 @@ package aws
 
 import (
 	"context"
+	"fmt"
+	"os"
 
 	"github.com/1Password/shell-plugins/sdk"
 	"github.com/1Password/shell-plugins/sdk/provision"
 	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+	confighelpers "github.com/99designs/aws-vault/v7/vault"
 )
 
 type awsProvisioner struct {
-	stsProvisioner    STSProvisioner
-	envVarProvisioner provision.EnvVarProvisioner
+	selectedProvisioner sdk.Provisioner
 }
 
 func AWSProvisioner() sdk.Provisioner {
-	return awsProvisioner{
-		envVarProvisioner: provision.EnvVarProvisioner{
-			Schema: defaultEnvVarMapping,
-		},
-	}
+	return awsProvisioner{}
 }
 
 func (p awsProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput, out *sdk.ProvisionOutput) {
-	totp, hasTotp := in.ItemFields[fieldname.OneTimePassword]
-	mfaSerial, hasMFASerial := in.ItemFields[fieldname.MFASerial]
-
-	if hasTotp && hasMFASerial {
-		p.stsProvisioner.MFASerial = mfaSerial
-		p.stsProvisioner.TOTPCode = totp
-		p.stsProvisioner.Provision(ctx, in, out)
-	} else {
-		p.envVarProvisioner.Provision(ctx, in, out)
+	provisioner := p.selectCorrespondingProvisioner(in.ItemFields, out)
+	if len(out.Diagnostics.Errors) > 0 {
+		return
 	}
+
+	provisioner.Provision(ctx, in, out)
 }
 
 func (p awsProvisioner) Deprovision(ctx context.Context, in sdk.DeprovisionInput, out *sdk.DeprovisionOutput) {
-	// Nothing to do here: environment variables get wiped automatically when the process exits.
+	p.selectedProvisioner.Deprovision(ctx, in, out)
 }
 
 func (p awsProvisioner) Description() string {
-	return p.envVarProvisioner.Description()
+	if p.selectedProvisioner != nil {
+		return p.selectedProvisioner.Description()
+	}
+	return "Provisions AWS with temporary credentials."
+}
+
+func (p *awsProvisioner) selectCorrespondingProvisioner(itemFields map[sdk.FieldName]string, out *sdk.ProvisionOutput) sdk.Provisioner {
+	awsConfig := getAWSConfiguration(out)
+	if len(out.Diagnostics.Errors) > 0 {
+		return nil
+	}
+
+	err := errorIfUnsupportedConfigurationIsPresent(awsConfig)
+	if err != nil {
+		out.AddError(err)
```

Comment:
These error messages don't mention that the unsupported config was detected in the AWS config file. Can we assume users are aware/remember that they've configured it there, or would it be worth mentioning that here in the error message?

Reply:
That configuration can also be from env vars not only from config file. I will state both possibilities for clarity.